h2/utilities.py

```diff
@@ -221,8 +221,8 @@ def _reject_te(headers, hdr_validation_flags):
     its value is anything other than "trailers".
     """
     for header in headers:
-        if header[0] == b'te':
-            if header[1].lower().strip() != b'trailers':
+        if header[0] in (b'te', u'te'):
+            if header[1].lower().strip() not in (b'trailers', u'trailers'):
```

Comment:
It is, yeah, so I think we can merge #281 *first* and then remove the `strip` call here.